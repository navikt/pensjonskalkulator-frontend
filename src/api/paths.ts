const HOST = import.meta.env.MODE === 'test' ? 'http://localhost:8088' : ''

export const PATH = `${import.meta.env.BASE_URL}api`

export const API_BASEURL = `${HOST}${PATH}`
