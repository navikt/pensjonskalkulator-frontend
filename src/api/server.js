// const express = require('express')
// const liveness = require('../src/__mocks__/data/liveness.json')
// const pensjonsberegning = require('../src/__mocks__/data/pensjonsberegning.json')

// const app = express()
// const port = 8088

// app.get('/api/status', (request, response) => {
//   response.send(liveness)
// })

// app.get('/api/pensjonsberegning', async (request, response) => {
//   response.send(pensjonsberegning)
// })

// app.listen(process.env.PORT || port, () => {
//   console.info('proxy server is running on port', process.env.PORT || port)
// })

// import { rest } from 'msw'
// import { setupServer } from 'msw/node'

import { setupServer } from 'msw/node'
import { handlers } from './handlers.js'

export const server = setupServer(...handlers)
server.listen({ onUnhandledRequest: 'error' })
