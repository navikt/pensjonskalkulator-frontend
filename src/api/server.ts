import { setupServer } from 'msw/node'
import { getHandlers } from './handlers.js'
import { rest } from 'msw'

export const API_TARGET = 'http://localhost:8088'
export const API_PATH = '/pensjon/kalkulator/api'

const handlers = getHandlers(`${API_TARGET}${API_PATH}`)
export const server = setupServer(...handlers)

server.listen({ onUnhandledRequest: 'error' })

type MockResponseOptions = {
  status?: number
  json?: any
}

export const mockResponse = (
  path: string,
  options: MockResponseOptions = {
    status: 200,
    json: 'OK',
  }
) => {
  server.use(
    rest.get(`${API_TARGET}${API_PATH}${path}`, (req, res, ctx) => {
      return res(ctx.status(options.status ?? 200), ctx.json(options.json))
    })
  )
}

export const mockErrorResponse = (
  path: string,
  options: MockResponseOptions = {
    status: 500,
    json: "Beep boop I'm an error!",
  }
) => {
  mockResponse(path, {
    ...options,
    status: options.status ?? 500,
  })
}
