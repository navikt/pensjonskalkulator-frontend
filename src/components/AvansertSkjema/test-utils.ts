export function getPreviousMonth(): string {
  const today = new Date()
  const currentMonth = parseInt(
    today.toLocaleDateString('en-CA').slice(5, 7),
    10
  ) // Extracts the month from the date string
  const previousMonth = currentMonth - 1
  return previousMonth.toString()
}
