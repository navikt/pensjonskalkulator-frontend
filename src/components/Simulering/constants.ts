export const MAX_UTTAKSALDER = 77
export const COLUMN_WIDTH = 25
export const TOOLTIP_YPOS = 35

export const SERIES_COLORS = {
  SERIE_COLOR_INNTEKT: 'var(--a-gray-500)',
  SERIE_COLOR_AFP: 'var(--a-purple-400)',
  SERIE_COLOR_TP: 'var(--a-green-400)',
  SERIE_COLOR_ALDERSPENSJON: 'var(--a-deepblue-500)',
  SERIE_COLOR_FADED_INNTEKT: 'var(--a-gray-300)',
  SERIE_COLOR_FADED_AFP: 'var(--a-purple-200)',
  SERIE_COLOR_FADED_TP: 'var(--a-green-200)',
  SERIE_COLOR_FADED_ALDERSPENSJON: 'var(--a-deepblue-200)',
}

export const SERIES_DEFAULT = {
  SERIE_INNTEKT: {
    type: 'column',
    pointWidth: COLUMN_WIDTH,
    name: 'Pensjonsgivende inntekt',
    color: SERIES_COLORS.SERIE_COLOR_INNTEKT,
  },
  SERIE_TP: {
    type: 'column',
    pointWidth: COLUMN_WIDTH,
    name: 'Pensjonsavtaler (arbeidsgivere m.m.)',
    color: SERIES_COLORS.SERIE_COLOR_TP,
  },
  SERIE_AFP: {
    type: 'column',
    pointWidth: COLUMN_WIDTH,
    name: 'AFP (Avtalefestet pensjon)',
    color: SERIES_COLORS.SERIE_COLOR_AFP,
  },
  SERIE_ALDERSPENSJON: {
    type: 'column',
    pointWidth: COLUMN_WIDTH,
    name: 'Alderspensjon (NAV)',
    color: SERIES_COLORS.SERIE_COLOR_ALDERSPENSJON,
  },
}

export const highchartsScrollingSelector = '.highcharts-scrolling'
