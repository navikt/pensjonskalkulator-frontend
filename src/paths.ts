export const getHost = (mode: string) =>
  mode === 'test' ? 'http://localhost:8088' : ''

// import.meta.env.BASE_URL defineres av base i vite.config.js = /pensjon/kalkulator
// legger til fallback for cypress som bruker webpack
export const API_PATH = import.meta.env
  ? `${import.meta.env.BASE_URL}/api`
  : '/pensjon/kalkulator/api'

export const HOST_BASEURL = `${getHost(import.meta.env?.MODE)}${
  import.meta.env?.BASE_URL
}`

export const API_BASEURL = `${getHost(import.meta.env?.MODE)}${API_PATH}`
