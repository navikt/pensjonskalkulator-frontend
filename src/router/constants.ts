export const BASE_PATH = '/pensjon/kalkulator'

export const externalUrls = {
  dinPensjon: 'https://nav.no/pensjon',
  dinPensjonBeholdning: 'https://www.nav.no/pensjon/opptjening/nb/',
  detaljertKalkulator: 'https://www.nav.no/pselv/simulering.jsf',
  alderspensjonsregler: 'https://www.nav.no/alderspensjon#beregning',
  afp: 'https://www.afp.no',
  afpPrivat: 'https://www.nav.no/afp-i-privat-sektor',
  garantipensjon: 'https://www.nav.no/minstepensjon',
  norskPensjon: 'https://norskpensjon.no/',
  uinnloggetKalkulator:
    'https://www.nav.no/pselv/simulering/forenkletsimulering.jsf',
  personvernerklaering:
    'https://www.nav.no/personvernerklaering#dine-rettigheter',
  personvernerklaeringKontaktOss:
    'https://www.nav.no/personvernerklaering#kontakt-nav',
  kontaktOss: 'https://www.nav.no/kontaktoss',
  planleggePensjon:
    'https://www.nav.no/planlegger-pensjon#noe-du-ikke-finner-svaret-p-her',
}

export const paths = {
  root: '/',
  login: '/login',
  henvisning: '/henvisning',
  start: '/start',
  samtykke: '/samtykke',
  utenlandsopphold: '/utenlandsopphold',
  offentligTp: '/offentlig-tp',
  afp: '/afp',
  samtykkeOffentligAFP: '/samtykke-offentlig-afp',
  ufoeretrygdAFP: '/ufoeretrygd-afp',
  sivilstand: '/sivilstand',
  uventetFeil: '/uventet-feil',
  beregningEnkel: '/beregning',
  beregningAvansert: '/beregning-detaljert',
  forbehold: '/forbehold',
  personopplysninger: '/personopplysninger',
} as const

export const henvisningUrlParams = {
  foedselsdato: '1963',
  apotekerne: 'apotekerne',
  utland: 'utland',
} as const
