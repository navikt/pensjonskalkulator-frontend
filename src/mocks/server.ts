import { rest } from 'msw'
import { setupServer } from 'msw/node'

import { API_BASEURL } from '@/paths'

import { getHandlers } from './handlers'

const handlers = getHandlers(API_BASEURL)
export const server = setupServer(...handlers)

type MockResponseOptions = {
  status?: number
  json?: ReturnType<typeof JSON.parse>
  text?: string
  method?: 'post' | 'get'
  baseUrl?: string
}

const defaultResponseOptions: Required<MockResponseOptions> = {
  status: 200,
  json: { data: 'OK' },
  method: 'get',
  baseUrl: `${API_BASEURL}`,
  text: '',
}

export const mockResponse = (
  path: string,
  inputOptions: MockResponseOptions = {}
) => {
  const options = { ...defaultResponseOptions, ...inputOptions }
  const parsedPath = `${options.baseUrl}${path}`

  server.use(
    rest[options.method](parsedPath, (_req, res, ctx) => {
      return res(
        ctx.status(options.status),
        options.text ? ctx.text(options.text) : ctx.json(options.json)
      )
    })
  )
}

const defaultErrorResponseOptions: Required<MockResponseOptions> = {
  status: 500,
  json: { data: "Beep boop I'm an error!" },
  text: '',
  method: 'get',
  baseUrl: `${API_BASEURL}`,
}

export const mockErrorResponse = (
  path: string,
  inputOptions: MockResponseOptions = {}
) => {
  const options = {
    ...defaultErrorResponseOptions,
    ...inputOptions,
  }
  mockResponse(path, options)
}
