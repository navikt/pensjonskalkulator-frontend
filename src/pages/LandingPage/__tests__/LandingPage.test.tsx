import { RouterProvider, createMemoryRouter } from 'react-router'
import { describe, it, vi } from 'vitest'

import { mockErrorResponse } from '@/mocks/server'
import { HOST_BASEURL } from '@/paths'
import { BASE_PATH, externalUrls, paths } from '@/router/constants'
import { routes } from '@/router/routes'
import { render, screen, userEvent, waitFor } from '@/test-utils'

const navigateMock = vi.fn()
vi.mock(import('react-router'), async (importOriginal) => {
  const actual = await importOriginal()
  return {
    ...actual,
    useNavigate: () => navigateMock,
  }
})

describe('LandingPage', () => {
  afterEach(() => {
    vi.clearAllMocks()
    vi.resetAllMocks()
  })

  it('har riktig sidetittel', async () => {
    const router = createMemoryRouter(routes, {
      basename: BASE_PATH,
      initialEntries: [`${BASE_PATH}${paths.login}`],
    })
    render(<RouterProvider router={router} />, {
      hasRouter: false,
    })
    await waitFor(async () => {
      expect(document.title).toBe('application.title')
    })
  })

  it('rendrer riktig innhold når brukeren er pålogget', async () => {
    const router = createMemoryRouter(routes, {
      basename: BASE_PATH,
      initialEntries: [`${BASE_PATH}${paths.login}`],
    })
    render(<RouterProvider router={router} />, {
      hasRouter: false,
    })

    await waitFor(() => {
      // Viser TopSection
      expect(
        screen.getByText('landingsside.for.deg.foedt.etter.1963')
      ).toBeVisible()
      expect(
        screen.getByText('landingsside.velge_mellom_detaljert_og_enkel')
      ).toBeVisible()
      // Viser riktig tekst på Enkel kalkulator knappen
      expect(
        screen.getByTestId('landingside-enkel-kalkulator-button').textContent
      ).toBe('landingsside.button.enkel_kalkulator')
      // Viser lenke til personopplysninger
      expect(
        screen.getByText('landingsside.link.personopplysninger')
      ).toBeVisible()
      expect(
        screen.getByText('landingsside.velge_mellom_detaljert_og_enkel_2')
      ).toBeVisible()
      // Viser riktig tekst på Detaljert kalkulator knappen
      expect(
        screen.getByTestId('landingside-detaljert-kalkulator-button')
          .textContent
      ).toBe('landingsside.button.detaljert_kalkulator')
    })
  })

  it('rendrer riktig innhold når brukeren ikke er pålogget', async () => {
    mockErrorResponse('/oauth2/session', {
      baseUrl: `${HOST_BASEURL}`,
    })

    const router = createMemoryRouter(routes, {
      basename: BASE_PATH,
      initialEntries: [`${BASE_PATH}${paths.login}`],
    })

    render(<RouterProvider router={router} />, {
      hasRouter: false,
    })

    await waitFor(() => {
      // Viser TopSection
      expect(
        screen.getByText('landingsside.for.deg.foedt.etter.1963')
      ).toBeVisible()
      expect(
        screen.getByText('landingsside.velge_mellom_detaljert_og_enkel')
      ).toBeVisible()
      // Viser riktig tekst på Enkel kalkulator knappen
      expect(
        screen.getByTestId('landingside-enkel-kalkulator-button').textContent
      ).toBe('landingsside.button.enkel_kalkulator_utlogget')
      // Viser lenke til personopplysninger
      expect(
        screen.getByText('landingsside.link.personopplysninger')
      ).toBeVisible()
      expect(
        screen.getByText('landingsside.velge_mellom_detaljert_og_enkel_2')
      ).toBeVisible()
      // Viser riktig tekst på Detaljert kalkulator knappen
      expect(
        screen.getByTestId('landingside-detaljert-kalkulator-button')
          .textContent
      ).toBe('landingsside.button.detaljert_kalkulator_utlogget')
      // Viser info for brukere født før 1963
      expect(
        screen.getByText('landingsside.for.deg.foedt.foer.1963')
      ).toBeVisible()
      expect(
        screen.getByText('landingsside.du.maa.bruke.detaljert')
      ).toBeVisible()
      // Viser riktig tekst på den tilleggsknappen som går til detaljert
      expect(
        screen.getByTestId('landingside-detaljert-kalkulator-second-button')
          .textContent
      ).toBe('landingsside.button.detaljert_kalkulator_utlogget')
      // Viser info om Uinnlogget kalkulator
      expect(
        screen.getByText('landingsside.text.uinnlogget_kalkulator')
      ).toBeVisible()
      expect(
        screen.getByText('landingsside.body.uinnlogget_kalkulator')
      ).toBeVisible()
      // Viser riktig tekst på Uinnlogget kalkulator knappen
      expect(
        screen.getByTestId('landingside-uinnlogget-kalkulator-button')
          .textContent
      ).toBe('landingsside.button.uinnlogget_kalkulator')
    })
  })

  it('går til detaljert kalkulator når brukeren klikker på detaljert kalkulator knappen', async () => {
    const user = userEvent.setup()

    const open = vi.fn()
    vi.stubGlobal('open', open)

    const router = createMemoryRouter(routes, {
      basename: BASE_PATH,
      initialEntries: [`${BASE_PATH}${paths.login}`],
    })
    render(<RouterProvider router={router} />, {
      hasRouter: false,
    })

    await waitFor(() => {
      expect(
        screen.getByTestId('landingside-detaljert-kalkulator-button')
      ).toBeDefined()
    })

    await user.click(
      screen.getByTestId('landingside-detaljert-kalkulator-button')
    )

    expect(open).toHaveBeenCalledWith(externalUrls.detaljertKalkulator, '_self')
  })

  it('går til enkel kalkulator når brukeren klikker på enkel kalkulator knappen', async () => {
    mockErrorResponse('/oauth2/session', {
      baseUrl: `${HOST_BASEURL}`,
    })

    const user = userEvent.setup()

    const open = vi.fn()
    vi.stubGlobal('open', open)

    const router = createMemoryRouter(routes, {
      basename: BASE_PATH,
      initialEntries: [`${BASE_PATH}${paths.login}`],
    })
    render(<RouterProvider router={router} />, {
      hasRouter: false,
    })

    await waitFor(async () => {
      await user.click(
        screen.getByTestId('landingside-enkel-kalkulator-button')
      )
    })

    expect(navigateMock).toHaveBeenCalledWith(`${paths.start}`)
  })

  it('går til detaljert kalkulator når brukeren klikker på tilleggsknappen i det andre avsnittet', async () => {
    mockErrorResponse('/oauth2/session', {
      baseUrl: `${HOST_BASEURL}`,
    })

    const user = userEvent.setup()

    const open = vi.fn()
    vi.stubGlobal('open', open)

    const router = createMemoryRouter(routes, {
      basename: BASE_PATH,
      initialEntries: [`${BASE_PATH}${paths.login}`],
    })
    render(<RouterProvider router={router} />, {
      hasRouter: false,
    })

    await waitFor(() => {
      expect(
        screen.getByTestId('landingside-detaljert-kalkulator-second-button')
      ).toBeDefined()
    })

    await user.click(
      screen.getByTestId('landingside-detaljert-kalkulator-second-button')
    )

    expect(open).toHaveBeenCalledWith(externalUrls.detaljertKalkulator, '_self')
  })

  it('går til uinnlogget kalkulator når brukeren klikker på uinnlogget kalkulator knappen', async () => {
    mockErrorResponse('/oauth2/session', {
      baseUrl: `${HOST_BASEURL}`,
    })

    const user = userEvent.setup()

    const open = vi.fn()
    vi.stubGlobal('open', open)

    const router = createMemoryRouter(routes, {
      basename: BASE_PATH,
      initialEntries: [`${BASE_PATH}${paths.login}`],
    })
    render(<RouterProvider router={router} />, {
      hasRouter: false,
    })

    await waitFor(() => {
      expect(
        screen.getByTestId('landingside-uinnlogget-kalkulator-button')
          .textContent
      ).toBe('landingsside.button.uinnlogget_kalkulator')
    })

    await waitFor(async () => {
      await user.click(
        screen.getByTestId('landingside-uinnlogget-kalkulator-button')
      )
    })

    expect(open).toHaveBeenCalledWith(
      externalUrls.uinnloggetKalkulator,
      '_self'
    )
  })
})
