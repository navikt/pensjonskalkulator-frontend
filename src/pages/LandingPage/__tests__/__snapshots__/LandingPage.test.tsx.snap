// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LandingPage > rendrer innlogget side 1`] = `
<DocumentFragment>
  <div
    class="_landingPage_6ae597"
  >
    <div
      class="navds-stack navds-vstack"
      style="--__ac-stack-wrap: nowrap; --__ac-stack-gap-xs: var(--a-spacing-10); --__ac-stack-direction-xs: column;"
    >
      <section>
        <div
          class="navds-stack navds-vstack"
          style="--__ac-stack-wrap: nowrap; --__ac-stack-gap-xs: var(--a-spacing-2); --__ac-stack-direction-xs: column;"
        >
          <h2
            class="navds-heading navds-heading--medium"
          >
            landingsside.for.deg.foedt.foer.1963
          </h2>
          <p
            class="navds-body-long navds-body-long--medium"
          >
            landingsside.du.maa.bruke.detaljert
          </p>
          <div>
            <button
              class="navds-button navds-button--secondary navds-button--medium"
              data-testid="landingside-detaljert-kalkulator-button"
            >
              <span
                class="navds-label"
              >
                landingsside.button.detaljert_kalkulator
              </span>
            </button>
          </div>
        </div>
      </section>
      <section>
        <div
          class="navds-stack navds-vstack"
          style="--__ac-stack-wrap: nowrap; --__ac-stack-gap-xs: var(--a-spacing-4); --__ac-stack-direction-xs: column;"
        >
          <h2
            class="navds-heading navds-heading--medium"
          >
            landingsside.for.deg.foedt.etter.1963
          </h2>
          <p
            class="navds-body-long navds-body-long--medium"
          >
            landingsside.velge_mellom_detaljert_og_enkel
          </p>
          <div>
            <p
              class="navds-body-long navds-body-long--medium"
            >
              landingsside.velge_mellom_detaljert_og_enkel_2
            </p>
            <ul>
              <li>
                landingsside.liste.1
              </li>
              <li>
                landingsside.liste.2
              </li>
              <li>
                landingsside.liste.3
              </li>
              <li>
                landingsside.liste.4
              </li>
            </ul>
          </div>
          <div
            class="navds-stack navds-hstack"
            style="--__ac-stack-wrap: wrap; --__ac-stack-gap-xs: var(--a-spacing-4); --__ac-stack-direction-xs: row;"
          >
            <button
              class="navds-button navds-button--secondary navds-button--medium"
            >
              <span
                class="navds-label"
              >
                landingsside.button.detaljert_kalkulator
              </span>
            </button>
            <button
              class="navds-button navds-button--secondary navds-button--medium"
              data-testid="landingside-enkel-kalkulator-button"
            >
              <span
                class="navds-label"
              >
                landingsside.button.enkel_kalkulator
              </span>
            </button>
          </div>
        </div>
      </section>
      <a
        href=""
      >
        landingsside.link.personopplysninger
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LandingPage > rendrer utlogget side 1`] = `
<DocumentFragment>
  <div
    class="_landingPage_6ae597"
  >
    <div
      class="navds-stack navds-vstack"
      style="--__ac-stack-wrap: nowrap; --__ac-stack-gap-xs: var(--a-spacing-10); --__ac-stack-direction-xs: column;"
    >
      <section>
        <div
          class="navds-stack navds-vstack"
          style="--__ac-stack-wrap: nowrap; --__ac-stack-gap-xs: var(--a-spacing-2); --__ac-stack-direction-xs: column;"
        >
          <h2
            class="navds-heading navds-heading--medium"
          >
            landingsside.for.deg.foedt.foer.1963
          </h2>
          <p
            class="navds-body-long navds-body-long--medium"
          >
            landingsside.du.maa.bruke.detaljert
          </p>
          <div>
            <button
              class="navds-button navds-button--secondary navds-button--medium"
              data-testid="landingside-detaljert-kalkulator-button"
            >
              <span
                class="navds-label"
              >
                landingsside.button.detaljert_kalkulator_utlogget
              </span>
            </button>
          </div>
        </div>
      </section>
      <section>
        <div
          class="navds-stack navds-vstack"
          style="--__ac-stack-wrap: nowrap; --__ac-stack-gap-xs: var(--a-spacing-4); --__ac-stack-direction-xs: column;"
        >
          <h2
            class="navds-heading navds-heading--medium"
          >
            landingsside.for.deg.foedt.etter.1963
          </h2>
          <p
            class="navds-body-long navds-body-long--medium"
          >
            landingsside.velge_mellom_detaljert_og_enkel
          </p>
          <div>
            <p
              class="navds-body-long navds-body-long--medium"
            >
              landingsside.velge_mellom_detaljert_og_enkel_2
            </p>
            <ul>
              <li>
                landingsside.liste.1
              </li>
              <li>
                landingsside.liste.2
              </li>
              <li>
                landingsside.liste.3
              </li>
              <li>
                landingsside.liste.4
              </li>
            </ul>
          </div>
          <div
            class="navds-stack navds-hstack"
            style="--__ac-stack-wrap: wrap; --__ac-stack-gap-xs: var(--a-spacing-4); --__ac-stack-direction-xs: row;"
          >
            <button
              class="navds-button navds-button--secondary navds-button--medium"
            >
              <span
                class="navds-label"
              >
                landingsside.button.detaljert_kalkulator_utlogget
              </span>
            </button>
            <button
              class="navds-button navds-button--secondary navds-button--medium"
              data-testid="landingside-enkel-kalkulator-button"
            >
              <span
                class="navds-label"
              >
                landingsside.button.enkel_kalkulator_utlogget
              </span>
            </button>
          </div>
        </div>
      </section>
      <section>
        <div
          class="navds-stack navds-vstack"
          style="--__ac-stack-wrap: nowrap; --__ac-stack-gap-xs: var(--a-spacing-2); --__ac-stack-direction-xs: column;"
        >
          <h2
            class="navds-heading navds-heading--medium"
            data-testid="uinlogget-kalkulator"
          >
            landingsside.text.uinnlogget_kalkulator
          </h2>
          <p
            class="navds-body-long navds-body-long--medium"
          >
            landingsside.body.uinnlogget_kalkulator
          </p>
          <div>
            <button
              class="navds-button navds-button--secondary navds-button--medium"
              data-testid="landingside-enkel-kalkulator-button"
            >
              <span
                class="navds-label"
              >
                landingsside.text.uinnlogget_kalkulator
              </span>
            </button>
          </div>
        </div>
      </section>
      <a
        href=""
      >
        landingsside.link.personopplysninger
      </a>
    </div>
  </div>
</DocumentFragment>
`;
